crates/primitives/src/scripts/metadata.rs

```diff
@@ -55,6 +57,38 @@ pub enum DepositMetadata {
 }
 
 impl<'tag> AuxiliaryData<'tag> {
+    /// Validates the OP_RETURN metadata and returns the metadata as bytes.
+    ///
+    /// This function validates:
+    /// - Tag size (must be 4 bytes)
+    /// - EE address size (must match expected size)
+    ///
+    /// Returns the metadata as bytes if validation passes.
+    pub fn validate_and_extract(
+        tag: &'tag [u8],
+        ee_address: &[u8],
+        expected_ee_address_size: usize,
+        metadata: DepositMetadata,
+    ) -> BridgeTxBuilderResult<Self> {
+        // Validate tag size
+        if tag.len() != 4 {
```

Comment:
I believe that the core side pads with 0 if the bytes, taken from params I suppose, are less then 4. The bridge should comply, or for the least the behavior should be consistent. It's fine if here we enforce it to be exactly 4 -- while I haven't checked the logic for where this function is being called, I assume that so far we haven't formatted the bytes in any specific way.

Reply:
[This](https://github.com/alpenlabs/alpen/pull/821/files#diff-e33e40a055c41e397078372c6f49f7246a07657733afeace44c761a94cb93cefR59-R65), no?

Reply:
oh, I see now that this isn't the right place to make my comment -- this is what we are taking from the tx. It's where we take value from the rollup params that this should be enforced.

Reply:
Yes it is done in 407a51f81fdf12de2cb9fed8ef2410addd87725a and the node panics, but we might not do that with a recent suggestion.